python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -318,12 +318,12 @@ def build_caching_repository_data_from_list(
 
     assets_without_keys = [ad for ad in assets_defs if not ad.keys]
     asset_graph = AssetGraph.from_assets(
-        [
+        {
```

Comment:
this is subtle and easy to miss visually, worth slapping a comment about what is going on here